eng/pipelines/docindex.yml

```diff
@@ -0,0 +1,189 @@
+resources:
+  repositories:
+    - repository: azure-sdk-tools
+      type: github
+      name: azure/azure-sdk-tools
+      ref: refs/heads/master
+      endpoint: azure
+
+jobs:
+  - job: CreateDocIndex
+    variables:
+      - template: templates/variables/globals.yml
+    pool:
+      vmImage: windows-2019
+    steps:
+      - pwsh: |
+          Invoke-WebRequest -Uri "https://github.com/dotnet/docfx/releases/download/v2.43.2/docfx.zip" `
+          -OutFile "docfx.zip" | Wait-Process; Expand-Archive -Path "docfx.zip" -DestinationPath "./docfx/"
+        workingDirectory: $(Build.BinariesDirectory)
+        displayName: Download and Extract DocFX
+      - task: CmdLine@2
+        displayName: Provision DocFX Directory
+        inputs:
+          script: $(Build.BinariesDirectory)/docfx/docfx.exe init -q
+          workingDirectory: $(Build.SourcesDirectory)
+          failOnStderr: true
+      - pwsh: New-Item -Path . -Name "templates" -ItemType "directory"
+        displayName: Create Template Directory
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+      - task: CopyFiles@2
+        displayName: Copy Template to templates folder
+        inputs:
+          sourceFolder: $(Build.SourcesDirectory)/eng/docgeneration/templates
+          content: '**\*'
+          targetFolder: $(Build.SourcesDirectory)/docfx_project/templates
+          overWrite: true
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/
+      - pwsh: ls
+        workingDirectory: $(Build.SourcesDirectory)/docfx_project/templates
+      - pwsh: |
+          Copy-Item "$(Build.SourcesDirectory)/eng/docgeneration/docfx.json" -Destination "$(Build.SourcesDirectory)/docfx_project/" -Force
+        displayName: Copy over docfx.json
+      - pwsh: |
+
```

Comment:
I feel like this should be an external powershell hosted in the `eng/` directory or the like. Maybe `eng/docgeneration`. 

However, I think that's the case for this entire thing, and both `JS` and `.NET` have the same problem. So I don't want to block this checkin based on this minor issue.

Reply:
@scbedd That's a fair enough request. I'll rework this when you file the issue.